yapapi/services/cluster.py

```diff
@@ -133,21 +133,26 @@ def spawn_instances(
     ) -> None:
         """Spawn new instances within this :class:`Cluster`.
 
-        :param num_instances: optional number of service instances to run. Defaults to a single
-            instance, unless `instance_params` is given, in which case, the :class:`Cluster` will spawn
-            as many instances as there are elements in the `instance_params` iterable.
-            if `num_instances` is not None and < 1, the method will immediately return and log a warning.
-        :param instance_params: optional list of dictionaries of keyword arguments that will be passed
-            to consecutive, spawned instances. The number of elements in the iterable determines the
-            number of instances spawned, unless `num_instances` is given, in which case the latter takes
+        :param num_instances: optional number of service instances to run. Defaults to
+            a single instance, unless `instance_params` is given, in which case,
```

Comment:
I think some explicit statement "num_instances, if passed, always takes precedence over `instance_params`", could be useful - this is not super-clear now (I understand this is written in `instance_params` doc, but the line I'm commenting now somehow contradicts this).

But: I'm not sure, just noting my minor confusion.

Reply:
I don't think there's contradiction ... I'd say the "unless" clause is there specifically to show that it _only_ defaults to 1 if we did not provide `instance_params` ... and the explanation is in the same sentence after "in which case"...